### PR TITLE
Feat/new map renderer

### LINKED ISSUE #4220: Refactor Map Renderer
The current implementation:

- is not scalable (it is almost impossible to add anything new so it blocks potential new features such as favorites)
- performs really badly as it is based on `Unity UI` and there are too many elements to rebuild when the map is dragged
- does not follow any good design practices: the flow is a callback hell, hard to follow, non-testable, etc

We need to refactor it to mitigate all this issues:

The HLD is described here: https://miro.com/app/board/uXjVPsG7NHc=/

Acceptance Criteria:
- [ ] Follow the HLD to implement a robust flexible system

### PR DESCRIPTION
**Pull request by mikhail-dcl**
## What does this PR change?

Closes #4220

1. Map Renderer was rewritten from scratch
2. The core idea behind it is using a camera and Render Texture to avoid canvases rebuilding
3. It's assumed that it is possible to change the implementation of the parts independently based on potential future improvements such as rendering multiple parts with a single shader
4. I will conduct a dedicated session to present the architectural solutions in detail
5. Introduced a test scene with a control panel to manipulate with Map Renderer API directly and independently, I didn't invest much time into its structure as it's for the testing editor purposes only
6. All map components are moved to a separate addressable
7. Introduced `InitializeAsync` for services
8. Properly implemented `IEquatable` for `MinimapMetadata
9. `HotScenesController` was refactored to comply with async API and follow better practices

For reviewers:

1. The epic was reviewed step-by-step so no thorough CR is required: all code already was validated. Though you can scan the code to familiarize yourself.
2. You can pay attention to the code outside of the map renderer itself that was required to modify

...

## How to test the changes?
1. Performance in Navigation Map should be much better even on the slower machines
2. The overall experience must be much smoother, there should be no spikes caused by dragging and zooming cameras
3. Slight visual degradation is acceptable, especially on lower resolution and distant zoom (when elements are smaller) due to the other rendering technique
4. Functionality of the nav map and minimap should be no worse than before refactoring; try to compare in case I missed something
5. I fixed parcels on NE (extra parcels outside of the square) so now they are accessible from the Nav Map 
6. In general just test the nav map and Minimap as they were a new feature

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Comment by popuz** [comment]
Found, that spawining in GPlaza didn't show proper name 
image

**Comment by dalkia** [comment]
> Found, that spawining in GPlaza didn't show proper name (only after changing position) image

To me, on first teleport, it worked fine 

image

But, going in and out of worlds generates inconsistent behaviour (wrong names, duplicate POIs icon). This was a bug also present the previous Map iteration

image

image

**Comment by dalkia** [comment]
There is a very nice performance increase while using the NavMap (first is `PR`, second is `prd`). I ran the tool while scrolling and zooming around.
 
Screen Shot 2023-04-02 at 23 04 44


Screen Shot 2023-04-02 at 23 07 10

**Comment by mikhail-dcl** [comment]
[MapRenderer Profiling - Still vs Dragging.zip](https://github.com/decentraland/unity-renderer/files/11135872/MapRenderer.Profiling.-.Still.vs.Dragging.zip)
Profiling

**Review by dalkia — APPROVED** [review]

**Review by Kinerius — APPROVED** [review]
same issue found with worlds, everything else works fine, gj!

**Review by kwssbocian — APPROVED** [review]
✔️Significant performance gains observed while using the world map.
✔️No issues observed regarding the functionality of the world map as well as the minimap.
